milk-planning/src/main/scala/dev/atedeg/mdm/milkplanning/dto/DTOs.scala

```diff
@@ -0,0 +1,25 @@
+package dev.atedeg.mdm.milkplanning.dto
+
+import dev.atedeg.mdm.milkplanning.{ Quantity, QuintalsOfMilk, RequestedProduct }
+import dev.atedeg.mdm.milkplanning.IncomingEvent.ReceivedOrder
+import dev.atedeg.mdm.milkplanning.OutgoingEvent.OrderMilk
+import dev.atedeg.mdm.products.dto.ProductDTO
+import dev.atedeg.mdm.utils.serialization.DTO
+import dev.atedeg.mdm.utils.serialization.DTOGenerators.*
+import dev.atedeg.mdm.utils.serialization.DTOOps.*
+
+final case class ReceivedOrderDTO(products: List[RequestedProductDTO])
+final case class RequestedProductDTO(product: ProductDTO, quantity: Int, requiredBy: String)
+final case class OrderMilkDTO(quintals: Int)
+
+object ReceivedOrderDTO:
+  import dev.atedeg.mdm.milkplanning.dto.RequestedProductDTO.given
+  given DTO[ReceivedOrder, ReceivedOrderDTO] = interCaseClassDTO
+
+object RequestedProductDTO:
+  given DTO[RequestedProduct, RequestedProductDTO] = interCaseClassDTO
+  private given DTO[Quantity, Int] = caseClassDTO
```

Comment:
Since the `RequestedProductDTO` is only needed by the `ReceiveOrderDTO` you can also put it as a private instance inside the `ReceivedOrderDTO` object:
```suggestion
object ReceivedOrderDTO:
  given DTO[ReceivedOrder, ReceivedOrderDTO] = interCaseClassDTO
  private given DTO[RequestedProduct, RequestedProductDTO] = interCaseClassDTO
  private given DTO[Quantity, Int] = caseClassDTO
```

That's how we did it in other bounded contexts